File: src/Applications/Orbital.Web/Orbital.Forecast.BlazorApplication/BlazorApp.Client/Library/DynamicLinkLibraries/Events/Event.Interfaces/ITimer.cs
using System;

namespace Event.Interfaces
{
    public interface ITimer
    {
        /// <summary>
        /// Time span
        /// </summary>
        TimeSpan TimeSpan
        {
            get;
        }

        /// <summary>
        /// Is enabled
        /// </summary>
        bool IsEnabled
        {
            get;
            set;
        }

        /// <summary>
        /// Event
        /// </summary>
        event Action Event;
    }
}

File: src/TypeScript/TypeScriptBackup/src/Library/Interfaces/ITimer.ts
import type { TimeSpan } from "../Utilities/DateTime/TimeSpan";
import type { IActionAddRemove } from "./IActionAddRemove";
import type { IActionT } from "./IActionT";

export interface ITimer  {

    getTimerTimeSpan(): TimeSpan

    isTimerEnabled(): boolean

    setTimerEnabled(enabled: boolean): void

    getTimerEvent(): IActionAddRemove

    setTimerEventT(action: IActionT<number>): void

}


